File: tests/test-minecraft/src/main/java/pize/tests/minecraft/game/gui/screen/screens/IOptionsScreen.java
package pize.tests.minecraft.game.gui.screen.screens;

import pize.tests.minecraft.game.Session;
import pize.tests.minecraft.game.gui.screen.Screen;
import pize.tests.minecraft.game.options.Options;

public abstract class IOptionsScreen extends Screen{

    public IOptionsScreen(Session session){
        super(session);
    }


    @Override
    public boolean shouldCloseOnEsc(){
        return true;
    }

    @Override
    public boolean renderDirtBackground(){
        return true;
    }

    @Override
    public void dispose(){ }

    public void saveOptions(){
        session.getOptions().save();
    }
    
    public Options getOptions(){
        return session.getOptions();
    }

}

File: tests/test-minecraft/src/main/kotlin/pize/tests/minecraft/game/gui/screen/screens/IOptionsScreen.kt
package pize.tests.minecraft.game.gui.screen.screens

import pize.tests.minecraft.game.Session
import pize.tests.minecraft.game.gui.screen.Screen
import pize.tests.minecraft.game.options.Options

abstract class IOptionsScreen(session: Session) : Screen(session) {
    override fun shouldCloseOnEsc(): Boolean {
        return true
    }

    override fun renderDirtBackground(): Boolean {
        return true
    }

    override fun dispose() {}
    fun saveOptions() {
        session.options.save()
    }

    val options: Options?
        get() = session.options
}
